### PR TITLE
AzureRT S55 PR #3 - Enable Disk Encryption Options for VM Creation

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by yugangw-msft** [comment]
@huangpf , please resolve the conflicts with the current dev, and i will merge

**Comment by huangpf** [comment]
@yugangw-msft OK. Updated.

File: lib/commands/arm/vm/vm._js [hunk 1]
```diff
@@ -101,6 +101,10 @@ exports.init = function (cli) {
     .option('--boot-diagnostics-storage-uri <boot-diagnostics-storage-uri>', $('The storage account URL for boot diagnostics. ' +
             'In the format of \'https://your_stoage_account_name.blob.core.windows.net/\'.'))
     .option('--disable-bginfo-extension', $('The switch parameter to disable the BGInfo extension for VM.'))
+    .option('--encryption-key-vault-id <disk-encryption-key-vault-id>', $('The OS disk encryption key vault ID.'))
```
Comment (SudhakaraReddyEvuri) [inline lib/commands/arm/vm/vm._js]: Please change the parameter name to ```disk-encryption-keyvault-id```

File: lib/commands/arm/vm/vmStorageProfile._js [hunk 2]
```diff
@@ -245,11 +246,40 @@ __.extend(VMStorageProfile.prototype, {
...
                  uri: null
-               }
+               },
+               encryptionSettings: null
              },
              isVhdParamAUrl: false
            };
 
+          // Encryption
+          if (this.params.diskEncryptionKeyVaultId || this.params.diskEncryptionKeySecretUrl) {
```
Comment (SudhakaraReddyEvuri) [inline lib/commands/arm/vm/vmStorageProfile._js]: Both DiskEncryptionKeyVaultID and DiskEncryptionKeyUrl are needed to enable encryption. KeyEncryptionkeyUrl and KeyEncryptionKeyVaultId should be used only when DiskEncryptionKey is valid.

**Pull request merged**